include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -161,46 +172,116 @@ std::optional<typename MapType::mapped_type> contains_key(const MapType& map, co
  * parameters and checking the operational status of the layout for each parameter combination. The operational domain
  * is then defined as the set of all parameter combinations for which the layout is operational. Different techniques
  * for performing these sweep are implemented.
- *
- * @tparam Key The type representing the key. Defaults to `parameter_point`.
- * @tparam Value The type representing the value. Defaults to `operational_status`.
  */
-template <typename Key = parameter_point, typename Value = operational_status>
-struct operational_domain
+class operational_domain : public sidb_simulation_domain<parameter_point, operational_status>
 {
+  public:
+    /**
+     * Sets the dimensions to sweep over, ordered by priority. The first dimension is the x dimension, the second
+     * dimension is the y dimension, etc.
+     *
+     * @param dims The dimensions to sweep over.
+     */
+    void set_dimensions(const std::vector<sweep_parameter>& dims)
+    {
+        dimensions = dims;
+    }
+    /**
+     * Adds a dimension to sweep over. The first dimension is the x dimension, the second dimension is the y dimension,
+     * etc.
+     *
+     * @param dim The dimension to add.
+     */
+    void add_dimension(const sweep_parameter& dim)
+    {
+        dimensions.push_back(dim);
+    }
+    /**
+     * Returns the dimensions to sweep over, ordered by priority. The first dimension is the x dimension, the second
+     * dimension is the y dimension, etc.
+     *
+     * @return The dimensions to sweep over.
+     */
+    [[nodiscard]] const std::vector<sweep_parameter>& get_dimensions() const noexcept
+    {
+        return dimensions;
+    }
+    /**
+     * Returns the number of dimensions to sweep over.
+     *
+     * @return The number of dimensions to sweep over.
+     */
+    [[nodiscard]]
+    std::size_t get_number_of_dimensions() const noexcept
+    {
+        return dimensions.size();
+    }
```

Comment:
`size()`?

Reply:
is already used for `` domain_values.size();`` in ``sidb_simulation_domain``.